Patch-Source: https://github.com/MusicPlayerDaemon/MPD/commit/59792cb0b801854ee41be72d33db9542735df754
removed news
---
From 59792cb0b801854ee41be72d33db9542735df754 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max.kellermann@gmail.com>
Date: Wed, 27 Jul 2022 10:59:48 +0200
Subject: [PATCH] decoder/ffmpeg: wrap FFmpeg include in "extern C"

Commit ebae25d175eb31 added that #include, but forgot to wrap it in
"extern C", so the linker tried to look up C++ symbols, causing linker
failure.

Closes https://github.com/MusicPlayerDaemon/MPD/issues/1582
---
 NEWS                             | 2 ++
 src/decoder/plugins/FfmpegIo.cxx | 5 ++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/decoder/plugins/FfmpegIo.cxx b/src/decoder/plugins/FfmpegIo.cxx
index 2e22d95991..5b5c8b40e3 100644
--- a/src/decoder/plugins/FfmpegIo.cxx
+++ b/src/decoder/plugins/FfmpegIo.cxx
@@ -21,10 +21,13 @@
 #define __STDC_CONSTANT_MACROS
 
 #include "FfmpegIo.hxx"
-#include "libavutil/mem.h"
 #include "../DecoderAPI.hxx"
 #include "input/InputStream.hxx"
 
+extern "C" {
+#include <libavutil/mem.h>
+}
+
 AvioStream::~AvioStream()
 {
 	if (io != nullptr) {
